src/common/eeprom.cpp

```diff
@@ -409,63 +413,115 @@ static eeprom_vars_t &eeprom_startup_vars() {
     return ret;
 }
 
+/**
+ * @brief reads eeprom record from RAM structure into variant variable
+ *
+ * @param id eeprom record index
+ * @return variant8_t eeprom record
+ */
 variant8_t eeprom_get_var(enum eevar_id id) {
-    uint16_t addr;
-    uint16_t size;
-    uint16_t data_size;
-    void *data_ptr;
     variant8_t var = variant8_empty();
     if (id < EEPROM_VARCOUNT) {
-        eeprom_lock();
         var = variant8_init(eeprom_map[id].type, eeprom_map[id].count, 0);
-        size = eeprom_var_size(id);
-        data_size = variant8_data_size(&var);
-        if (size == data_size) {
-            addr = eeprom_var_addr(id);
-            data_ptr = variant8_data_ptr(&var);
-            st25dv64k_user_read_bytes(addr, data_ptr, size);
+        eeprom_get_var(id, variant8_data_ptr(&var), variant8_data_size(&var));
+    }
+    return var;
+}
+
+/**
+ * @brief reads eeprom record from RAM structure
+ *
+ * @param id eeprom record index
+ * @param var_ptr pointer to variable to be read
+ * @param var_size size of variable
+ */
+static void eeprom_get_var(enum eevar_id id, void *var_ptr, size_t var_size) {
+    if (id < EEPROM_VARCOUNT) {
+        size_t size = eeprom_var_size(id);
+        if (size == var_size) {
+            eeprom_vars_t &vars = eeprom_startup_vars();
+            const void *var_addr = eeprom_var_ptr(id, vars);
+            // need to lock even if it is not accesing eeprom
+            // because RAM structure changes during write also
+            eeprom_lock();
+            memcpy(var_ptr, var_addr, size);
+            eeprom_unlock();
         } else {
             //TODO:error
             log_error(EEPROM, "%s: invalid data size", __FUNCTION__);
         }
-        eeprom_unlock();
     }
-    return var;
 }
 
+/**
+ * @brief function that writes variant8_t to eeprom, also actualizes crc and RAM structure
+ *
+ * @param id eeprom record index
+ * @param var variant variable holding data to be written
+ */
 void eeprom_set_var(enum eevar_id id, variant8_t var) {
-    uint16_t addr;
-    uint16_t size;
-    uint16_t data_size;
-    void *data_ptr;
     if (id < EEPROM_VARCOUNT) {
+        if (variant8_get_type(var) == eeprom_map[id].type) {
+            eeprom_set_var(id, variant8_data_ptr(&var), eeprom_var_size(id));
+        } else {
+            // TODO: error
+            log_error(EEPROM, "%s: variant type missmatch on id: %x", __FUNCTION__, id);
+        }
+
+    } else {
+        assert(0 /* EEProm var Id out of range */);
```

Comment:
You use assert(0) here, but in eeprom_get_var(enum eevar_id id) nothing happens when id is out of range. So this is little bit inconsistent.

Reply:
It just moved code, ok I will fix it